command/analyzer/run/run.go

```diff
@@ -0,0 +1,208 @@
+			AnalysisCommand:      analyzerTOMLData.Analysis.Command,
+			ContainerCodePath:    containerCodePath,    // /code
+			ContainerToolBoxPath: containerToolBoxPath, // /toolbox
+		},
+	}
+
+	// Building the Analyzer image
+	fmt.Println("Building Analyzer image...")
+	if err := d.BuildAnalyzerDockerImage(); err != nil {
+		return err
+	}
+
+	// Resolve the path of source code to be analyzed based on the user input
+	d.AnalysisOpts.HostCodePath, err = a.resolveAnalysisSourcePath()
+	if err != nil {
+		return err
+	}
+
+	/* Prepare the analysis_config.json here and mount into the container at `TOOLBOX_PATH/analysis_config.json`
+	 * The analysis_config.json will have path prepended with the CODE_PATH of the container and not local CODE_PATH */
+	analysisRun := analysis_config.AnalysisRun{
+		AnalyzerName:      analyzerTOMLData.Name,
+		LocalCodePath:     d.AnalysisOpts.HostCodePath,
+		ContainerCodePath: containerCodePath,
+	}
+
+	analysisConfig, err := analysisRun.ConfigureAnalysis()
+	if err != nil {
+		log.Println(err)
+	}
+
+	// Modify the paths of analysis_config.json file to use the container based CODE_PATH instead
+	// of the local CODE_PATH
+	modifyAnalysisConfigFilepaths(analysisConfig, d.AnalysisOpts.HostCodePath, d.AnalysisOpts.ContainerCodePath)
+
+	// Write the analysis_config.json to TOOLBOX_PATH
+	if err = a.writeAnalysisConfig(analysisConfig); err != nil {
+		return err
+	}
+	d.AnalysisOpts.HostToolBoxPath = a.TempToolBoxDirectory
+
+	// Write the analysis_config data into a temp /toolbox directory mount it as well
+	if err = d.StartDockerContainer(); err != nil {
+		return err
+	}
+
+	// Reads the LSP based analysis result written in the `analysis_results.json` file present in
+	// the directory of the Analyzer
+	analysisResultsData, err := os.ReadFile("analysis_results.json")
+	if err != nil {
+		return err
+	}
+
+	// Unmarshal the analysis result data present in the LSP format
+	analysisResultContent := analysis_config.AnalysisResult{}
+	if err = json.Unmarshal(analysisResultsData, &analysisResultContent); err != nil {
+		return err
+	}
+
+	// Format the LSP based results to the default format supported by DeepSource
+	analysisResult := analysisRun.FormatLSPResultToDefault(&analysisResultContent)
+
+	res, _ := json.Marshal(analysisResult)
+	os.WriteFile("analysis_results_default.json", res, 0o644)
+
+	return nil
+}
+
+/* Resolves the code to be analyzed by the Analyzer.
+ * The user passes it as an argument to the command `deepsource analyzer run <directory/repository URL>`
+ * Parse the argument and check if its a URL, if not then resolve the local directory path */
+func (a *AnalyzerRunOpts) resolveAnalysisSourcePath() (string, error) {
+	var err error
+
+	// Check if the source path is a valid VCS URL
+	if isValidUrl(a.SourcePath) {
+		a.RemoteSource = true
+
+		// Clone the repository to a temporary directory
+		// TODO: Check here if /tmp directory actually exists. Also, take care of Windows please?
```

Comment:
`os.MkdirTemp` [makes use of `TempDir`](https://cs.opensource.google/go/go/+/refs/tags/go1.18.2:src/os/tempfile.go;l=80-82) for creating the directory.

`TempDir` [internally handles](https://cs.opensource.google/go/go/+/refs/tags/go1.18.2:src/os/file.go;l=379-383;drc=654b03167f7d887d09a2d25bb3d1b4db9c8ae4f8) Windows temp directories too :D

Reply:
Hey @burntcarrot 👋 , thanks for the heads up 🙌